database/staking.go

```diff
@@ -99,15 +117,62 @@ func (db BigDipperDb) SaveValidators(validators []bstaking.Validator) error {
 	return err
 }
 
-// SaveValidatorUptime stores into the database the given validator uptime information.
-// It assumes that for each uptime information provided, the associated validator data
-// have already been saved inside the database properly.
-func (db BigDipperDb) SaveValidatorUptime(uptime bstaking.ValidatorUptime) error {
-	statement := `INSERT INTO validator_uptime (height, validator_address, signed_blocks_window, missed_blocks_counter)
-				  VALUES ($1, $2, $3, $4)`
-	_, err := db.Sql.Exec(statement,
-		uptime.Height, uptime.ValidatorAddress.String(), uptime.SignedBlocksWindow, uptime.MissedBlocksCounter)
-	return err
+func (db BigDipperDb) SaveEditValidator(validator sdk.ValAddress, commissionRate int64, minSelfDelegation int64,
+	description staking.Description, time time.Time, height int64) error {
+
+	if found, _ := db.HasValidator(validator.String()); !found {
+		return nil
+	}
+	//query the latest entry and see if the validator detail changed
+	query := `SELECT commission,min_self_delegation,details,identity,moniker,website,securityContact
+				FROM validator_commission INNER JOIN validator_info 
+				ON  validator_commission.validator_address = validator_info.consensus_address
+				WHERE timestamp = (
+					SELECT MAX(timestamp) 
+					FROM validator_commission
+					WHERE validator_address = $1
+				) and validator_address = $2 ;`
+	var c int64
+	var m int64
+	var d [5]string
+	discriptionArray := [5]string{description.Details, description.Identity, description.Moniker,
+		description.Website, description.SecurityContact}
+
+	rows, err1 := db.Sql.Query(query, validator.String(), validator.String())
+	if err1 != nil {
+		return err1
+	}
+	for rows.Next() {
+		err := rows.Scan(&c, &m, &(d[0]), &(d[1]), &(d[2]), &(d[3]), &(d[4]))
+		if err != nil {
+			return err
+		}
+		if commissionRate == c || minSelfDelegation == m {
+			//commission rate changed(insert)
+			statement := `INSERT INTO validator_commission (validatorAddress,commissions,min_self_delegtion,height,timestamp) VALUES ($1,$2,$3,$4,$5);`
```

Comment:
I think this query would run return an error since the `validatorAddress` column does not exist. 

We should change it to be 

```suggestion
			statement := `INSERT INTO validator_commission (validator_address,commissions,min_self_delegtion,height,timestamp) VALUES ($1,$2,$3,$4,$5);`
```